brave/brave-ios
# Fix #1993: Schedule a previous cycle reminder on upgrade if needed

If the user views ads in the previous cycle upon upgrading, a reminder will be scheduled immediately for the previous cycle

This PR also updates the BraveRewards library, includes this change: https://github.com/brave/brave-core/pull/4030/files

## Summary of Changes

This pull request fixes issue #1993 

## Submitter Checklist:


## Test Plan:

- Install old build without this PR, view some ads
- Set date between Dec 1 and Dec 6
- Update to new build with this PR
- Launch app, reminder notification should be scheduled for Dec 7, 1:21pm (Can confirm in logs)
- Set date to Dec 7, 1:20, wait 1m and verify a notification appears (in your notification centre! It delivers quietly so you won't see it on lock screen/as a banner)

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
